tests/test-inapp-receipts.js

```diff
@@ -48,13 +47,13 @@ describe('fxpay.init(): receipts', function() {
           assert.equal(info.productId, helper.apiProduct.guid);
           assert.equal(info.name, helper.apiProduct.name);
           assert.equal(info.smallImageUrl, helper.apiProduct.logo_url);
+          assert.equal(info.receiptInfo.status, 'ok');
         }
         done(err);
       }
     });
 
-    helper.appSelf.onsuccess();
-    helper.server.respond();
+    validator.finish();
```

Comment:
I'm not sure what you mean. Both of these lines just moved down into the ``finish()`` function: https://github.com/mozilla/fxpay/pull/72/files#diff-d5ebdbfb88cc7626c6e2f1d9f32e6d5aR207

The ``onsuccess()`` is needed to resolve the faux DOMRequest. Also, I hate testing async code :syringe: